sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -555,13 +566,15 @@ def run_pipeline(self, pipeline, options):
 
     self.job = apiclient.Job(options, self.proto_pipeline)
 
-    # Dataflow Runner v1 requires output type of the Flatten to be the same as
-    # the inputs, hence we enforce that here. Dataflow Runner v2 does not
-    # require this.
-    pipeline.visit(self.flatten_input_visitor())
+    # TODO: Consider skipping these for all use_portable_job_submission jobs.
```

Comment:
We're migrating all "ordinary" dataflow pipelines to portable_job_submission. In that case the `pipeline` object is provided instead of the `pipeline_proto` which is what the TODO is about.

Reply:
Cool - thanks for clarifying!